### PR TITLE
Add SetupRemarketing example

### PR DESCRIPTION
**Pull request by devchas**
This code example is for an upcoming guide on remarketing.

### CODE REVIEW

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @param campaignId the campaign under which to remove the ad group criteria.
+   */
+  private void removeExistingListCriteriaFromAdGroup(
+      GoogleAdsClient googleAdsClient,
+      long customerId,
+      long campaignId) {
+    // Retrieves all of the ad group criteria under a campaign.
+    List<String> adGroupCriteria =
```
Comment (nwbirnie) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java]: Is this wrapping correct? Hard to see on Github. Would suggest `google-java-format -i`.

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+      // resource names to the list.
+      for (GoogleAdsRow googleAdsRow : searchPagedResponse.iterateAll()) {
+        String adGroupCriterionResourceName = googleAdsRow.getAdGroupCriterion().getResourceName();
+        System.out.printf(
+            "Ad group criterion with resource name '%s' was found.%n",
+            adGroupCriterionResourceName);
+        userListCriteria.add(adGroupCriterionResourceName);
+      }
+    }
+  return userListCriteria;
```
Comment (nwbirnie) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java]: Formatting. Maybe you want to configure Intellij to run google-java-format on all files? Then ctrl+shift+f formats the entire file (IIRC that's the default key combo).

http://screenshot/uPkTzLyGCH2

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java [hunk 3]
```diff
@@ -0,0 +1,451 @@
...
+    RuleBasedUserListInfo ruleBasedUserListInfo = RuleBasedUserListInfo.newBuilder()
+        // Optional: To include past users in the user list, set the prepopulation_status to
+        // REQUESTED.
+        .setPrepopulationStatus(UserListPrepopulationStatus.REQUESTED)
+        .setExpressionRuleUserList(expressionRuleUserListInfo)
+        .build();
+
+    // Creates the user list.
+    UserList userList = UserList.newBuilder()
+        .setName(StringValue.of("All visitors to example.com" + System.currentTimeMillis()))
```
[the remark this answers is not in the document]
Comment (jradcliff) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java]: @nwbirnie For consistency with existing examples, I propose we stay with `currentTimeMillis()`, and revisit this holistically as a separate effort. If we only use UUID here and not elsewhere, that could be confusing for users. We already have [issue 42](https://github.com/googleads/google-ads-java/issues/42) for this effort.

[the remark this answers is not in the document]
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/SetupRemarketing.java]: SGTM

**Pull request merged**